pkg/thing/interactors/update_data.go

```diff
@@ -0,0 +1,69 @@
+package interactors
+
+import (
+	"fmt"
+
+	"github.com/CESARBR/knot-babeltower/pkg/network"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+)
+
+var (
+	ErrNoSchema = fmt.Errorf("thing has no schema")
+	ErrDataInvalid = fmt.Errorf("data is incompatible with thing's schema")
+)
+
+// UpdateData executes the use case operations to update data in thing
+func (i *ThingInteractor) UpdateData(authorization, thingID string, data []network.Data) error {
+	err := i.verifyThingData(authorization, thingID, data)
+	if err != nil {
+		return fmt.Errorf("unable to update data: %w", err)
+	}
+
+	err = i.clientPublisher.SendUpdateData(thingID, data)
+	if err != nil {
+		return fmt.Errorf("error sending message to client: %w", err)
+	}
+
+	i.logger.Info("data update command successfully sent")
+	return nil
+}
+
+func (i *ThingInteractor) verifyThingData(authorization, thingID string, data []network.Data) error {
+	thing, err := i.thingProxy.Get(authorization, thingID)
+	if err != nil {
+		return fmt.Errorf("error getting thing metadata: %w", err)
+	}
+
+	if thing.Schema == nil {
+		return ErrNoSchema
+	}
+
+	for _, d := range data {
+		if !validateSchema(d, thing.Schema) {
+			return ErrDataInvalid
+		}
+	}
+
+	return nil
+}
+
+func validateSchema(data network.Data, schema []entities.Schema) bool {
+	for _, s := range schema {
+		if s.SensorID == data.SensorID {
+			switch data.Value.(type) {
+			case int:
+				return s.ValueType == 1 // valueType 1 => INT
+			case float64:
+				return s.ValueType == 2 // valueType 2 => FLOAT
+			case bool:
+				return s.ValueType == 3 // valueType 3 => BOOL
+			case string:
+				return s.ValueType == 4 // valueType 4 => RAW
```

Comment:
Just FYI, this is more of a opinionated comment :stuck_out_tongue_closed_eyes:  

Considering that `ValueType == 1` and `// valueType 1 =>` mean the same thing, I would avoid this redundancy and leave it only as `int`, `float`, `bool` and `raw` (as they're written in code: lowercase).

```suggestion
				return s.ValueType == 1 // int
			case float64:
				return s.ValueType == 2 // float
			case bool:
				return s.ValueType == 3 // bool
			case string:
				return s.ValueType == 4 // raw
```

Reply:
@lcbm Nice, I like it! Done :smile: